### DISCUSSION TITLE
Lesson%201%3A%20Hello%20Metal/3.%20Textures/

### DISCUSSION THREAD

**Discussion by giscus** [wmarti/MetalTutorial#3] [Announcements]
# Lesson%201%3A%20Hello%20Metal/3.%20Textures/

https://metaltutorial.com/Lesson%201%3A%20Hello%20Metal/3.%20Textures/

**Comment by Andreyogld3d** [comment]
Hi! thank you for Lesson.
I suggest you provide example with using Metal Argument Buffers(https://developer.apple.com/documentation/metal/buffers/about_argument_buffers) and MTLHeap(https://developer.apple.com/documentation/metal/mtlheap)
MTLRenderCommandEncoder::setFragmentTexture - Old GL style with slow performance

**Comment by PnTsoi** [comment]
I think there's a mistake, the file name should be `stb_image.cpp` and not `stbi_image.cpp`

**Comment by chuzcjoe** [comment]
For anyone who might be interested in pure C++ implementation of this tutorial:
https://github.com/chuzcjoe/CORE/blob/master/examples/metal/MetalTextureDemo/main.cpp
